library/ssl_tls13_generic.c

```diff
@@ -402,27 +402,31 @@ int mbedtls_ssl_write_signature_algorithms_ext( mbedtls_ssl_context *ssl,
 
 int mbedtls_ssl_parse_signature_algorithms_ext( mbedtls_ssl_context *ssl,
                                         const unsigned char *buf,
-                                        size_t len )
+                                        size_t buf_len )
 {
     size_t sig_alg_list_size; /* size of receive signature algorithms list */
     const unsigned char *p; /* pointer to individual signature algorithm */
-    const unsigned char *end = buf + len; /* end of buffer */
+    const unsigned char *end = buf + buf_len; /* end of buffer */
     const int *md_cur; /* iterate through configured signature schemes */
     int signature_scheme; /* store received signature algorithm scheme */
-    int got_common_sig_alg = 0;  /* record whether there is a match between configured and received signature algorithms */
-    size_t num_supported_hashes;
-    uint32_t i; /* iterature through received_signature_schemes_list */
+    size_t num_supported_hashes = 0;
+    uint32_t i = 0; /* iterate through received_signature_schemes_list */
```

Comment:
Wondering: should we rename `i` to something more meaningful, like `num_of_matched_hashes`?